src/qibocal/cli/auto_builder.py

```diff
@@ -0,0 +1,78 @@
+import datetime
+import os
+
+import yaml
+
+from qibocal.auto.execute import Executor
+
+from .builders import ActionBuilder, load_yaml
+
+
+class AutoCalibrationBuilder(ActionBuilder):
+    def __init__(self, runcard, folder=None, force=False):
+        super().__init__(runcard, folder, force)
+        self.executor = Executor.load(self.runcard)
+
+    def run(self):
+        if self.platform is not None:
+            self.platform.connect()
+            self.platform.setup()
+            self.platform.start()
+
+        self.executor.run(self.platform, self.qubits, self.folder)
+
+        if self.platform is not None:
+            self.platform.stop()
+            self.platform.disconnect()
+
+    def dump_report(self):
+        from qibocal.web.report import create_autocalibration_report
+
+        # update end time
+        meta = load_yaml(f"{self.folder}/meta.yml")
+        e = datetime.datetime.now(datetime.timezone.utc)
+        meta["end-time"] = e.strftime("%H:%M:%S")
+        with open(f"{self.folder}/meta.yml", "w") as file:
+            yaml.dump(meta, file)
+
+        create_autocalibration_report(self.folder, self.executor.history)
+
+
+class AutoCalibrationReportBuilder:
+    def __init__(self, path, history):
+        self.path = path
+        self.metadata = load_yaml(os.path.join(path, "meta.yml"))
+
+        # find proper path title
+        base, self.title = os.path.join(os.getcwd(), path), ""
+        while self.title in ("", "."):
+            base, self.title = os.path.split(base)
```

Comment:
Not sure I understand the logic, but I guess you want something like:
```suggestion
        self.title = path.name
```
Otherwise, just tell me which part of the path you're looking for here, and I can help you find a more explicit way of getting it :)
(one or two examples should be enough to understand)